<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Application</class>
 <widget class="QMainWindow" name="Application">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>828</width>
    <height>543</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>HonoursProject (B00176646)</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="margin" stdset="0">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QStackedWidget" name="stackedWidget_2">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="terrain_page">
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QWidget" name="widget_2" native="true">
          <property name="maximumSize">
           <size>
            <width>300</width>
            <height>16777215</height>
           </size>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="1" column="0">
            <layout class="QVBoxLayout" name="verticalLayout">
             <property name="spacing">
              <number>0</number>
             </property>
             <item>
              <widget class="QWidget" name="widget" native="true">
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <property name="topMargin">
                 <number>1</number>
                </property>
                <item>
                 <layout class="QVBoxLayout" name="verticalLayout_5">
                  <property name="topMargin">
                   <number>0</number>
                  </property>
                  <item>
                   <widget class="QListWidget" name="terrain_oplist">
                    <property name="toolTip">
                     <string>List of available terrain operations</string>
                    </property>
                    <property name="whatsThis">
                     <string>List of available terrain operations</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="terrain_add_op">
                    <property name="text">
                     <string>Add</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="terrain_remove_op">
                    <property name="text">
                     <string>Remove</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QVBoxLayout" name="verticalLayout_4">
                  <property name="topMargin">
                   <number>0</number>
                  </property>
                  <property name="rightMargin">
                   <number>0</number>
                  </property>
                  <item>
                   <widget class="QListWidget" name="terrain_active_oplist"/>
                  </item>
                  <item>
                   <widget class="QPushButton" name="terrain_up">
                    <property name="text">
                     <string>Up</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="terrain_down">
                    <property name="text">
                     <string>Down</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QProgressBar" name="terrain_progress">
               <property name="value">
                <number>0</number>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <property name="spacing">
                <number>6</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QPushButton" name="terrain_generate">
                 <property name="text">
                  <string>Generate Terrain</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="terrain_export">
                 <property name="text">
                  <string>Export</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item row="0" column="0">
            <widget class="QTreeView" name="propertyInspector">
             <property name="alternatingRowColors">
              <bool>false</bool>
             </property>
             <property name="selectionBehavior">
              <enum>QAbstractItemView::SelectRows</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QTabWidget" name="terrain_viewport_tab">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="tabPosition">
           <enum>QTabWidget::South</enum>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab">
           <attribute name="title">
            <string>3D View</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <layout class="QGridLayout" name="terrain_viewport_layout"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_2">
           <attribute name="title">
            <string>Image</string>
           </attribute>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="bsg_page">
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QFrame" name="frame">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>500</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <property name="spacing">
            <number>0</number>
           </property>
           <property name="margin" stdset="0">
            <number>0</number>
           </property>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QTextEdit" name="bsg_text"/>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <property name="spacing">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QPushButton" name="bsg_load">
                 <property name="text">
                  <string>Load</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="bsg_save">
                 <property name="text">
                  <string>Save</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="bsg_clear">
                 <property name="text">
                  <string>Clear</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="bsg_apply">
                 <property name="text">
                  <string>Apply</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="bsg_num_iterations">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>1000</number>
                 </property>
                 <property name="value">
                  <number>1</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="frame_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_12">
           <item row="0" column="0">
            <layout class="QGridLayout" name="bsg_viewport"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="texturing_page">
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QFrame" name="frame_7">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>500</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <property name="topMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QToolBox" name="toolBox">
               <property name="currentIndex">
                <number>0</number>
               </property>
               <widget class="QWidget" name="page">
                <property name="geometry">
                 <rect>
                  <x>0</x>
                  <y>0</y>
                  <width>98</width>
                  <height>89</height>
                 </rect>
                </property>
                <attribute name="label">
                 <string>Shader</string>
                </attribute>
                <layout class="QGridLayout" name="gridLayout_11">
                 <item row="0" column="0">
                  <widget class="QTextEdit" name="proctex_fragedit"/>
                 </item>
                </layout>
               </widget>
               <widget class="QWidget" name="page_2">
                <property name="geometry">
                 <rect>
                  <x>0</x>
                  <y>0</y>
                  <width>98</width>
                  <height>89</height>
                 </rect>
                </property>
                <attribute name="label">
                 <string>Compiler Output</string>
                </attribute>
                <layout class="QGridLayout" name="gridLayout_14">
                 <item row="0" column="0">
                  <widget class="QTextBrowser" name="proctext_errors">
                   <property name="maximumSize">
                    <size>
                     <width>16777215</width>
                     <height>16777215</height>
                    </size>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_9">
               <property name="topMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QPushButton" name="proctex_sac">
                 <property name="text">
                  <string>Save and Compile</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="proctex_load">
                 <property name="text">
                  <string>Load</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="proctex_new">
                 <property name="text">
                  <string>New</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="frame_5">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_13">
           <item row="0" column="0">
            <layout class="QGridLayout" name="proctex_viewport"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="game_page">
       <layout class="QGridLayout" name="gridLayout_16">
        <property name="margin" stdset="0">
         <number>0</number>
        </property>
        <property name="spacing">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <layout class="QGridLayout" name="game_viewport"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>828</width>
     <height>22</height>
    </rect>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::ActionsContextMenu</enum>
   </property>
   <widget class="QMenu" name="menuGoto">
    <property name="title">
     <string>Goto</string>
    </property>
    <addaction name="actionTerrain"/>
    <addaction name="actionBSG"/>
    <addaction name="actionTexturing"/>
    <addaction name="separator"/>
    <addaction name="gameBegin"/>
   </widget>
   <addaction name="menuGoto"/>
  </widget>
  <action name="actionTerrain">
   <property name="text">
    <string>Terrain</string>
   </property>
  </action>
  <action name="actionBSG">
   <property name="text">
    <string>Basic Shape Grammar</string>
   </property>
  </action>
  <action name="actionTexturing">
   <property name="text">
    <string>Texturing</string>
   </property>
  </action>
  <action name="gameBegin">
   <property name="text">
    <string>Game</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
